### PR TITLE
feat(cluster): remote step

### PR DESCRIPTION
**Pull request by bdebon**
# What does this PR do?

- Cluster creation remote step
- Cluster settings remote

CleanShot 2023-02-07 at 13 17 42@2x

---

## PR Checklist

- [ ] This PR introduces breaking change(s) and has been labeled as such
- [ ] This PR introduces new store changes
- [x] I have followed the library pattern i.e `feature`, `ui`, `data`, `utils`
- [x] I made sure the code is type safe (no any)

### CODE REVIEW

**Comment by evoxmusic** [comment]
A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [[PR] staging - feat(cluster): remote step - 2023-02-07T12:18:11Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/c41fbdb8-201a-461f-b9f8-504bd879fa47/services/general)

 Another comment will be posted when deployments are terminated

File: libs/shared/console-shared/src/lib/cluster-settings/ui/cluster-remote-settings/cluster-remote-settings.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import { wrapWithReactHookForm } from '__tests__/utils/wrap-with-react-hook-form'
+import { ClusterRemoteData } from '@qovery/shared/interfaces'
+import ClusterRemoteSettings, { ClusterRemoteSettingsProps } from './cluster-remote-settings'
+
+describe('ClusterRemoteSettings', () => {
+  let defaultValues: ClusterRemoteData
+  let props: ClusterRemoteSettingsProps
+  beforeEach(() => {
+    defaultValues = {
+      ssh_key: 'ssh_key dslkjsdflkjsdflksjdf',
```
Comment (RemiBonnet) [inline libs/shared/console-shared/src/lib/cluster-settings/ui/cluster-remote-settings/cluster-remote-settings.spec.tsx:12]: Can you use the mock here ?

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-remote/step-remote.spec.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const props: StepRemoteProps = {
+  onSubmit: jest.fn((e) => e.preventDefault()),
+}
+
+describe('StepRemote', () => {
+  let defaultValues: ClusterRemoteData
+
+  beforeEach(() => {
+    defaultValues = {
+      ssh_key: 'ssh_key dslkjsdflkjsdflksjdf',
```
Comment (RemiBonnet) [inline libs/pages/clusters/src/lib/ui/page-clusters-create/step-remote/step-remote.spec.tsx:16]: Can you use the mock here ?

File: libs/pages/cluster/src/lib/ui/page-settings-remote/page-settings-remote.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  loading: false,
+  onSubmit: jest.fn((e) => e.preventDefault()),
+}
+
+describe('PageSettingsRemote', () => {
+  let defaultValues: ClusterRemoteData
+
+  beforeEach(() => {
+    defaultValues = {
+      ssh_key: 'ssh_key dslkjsdflkjsdflksjdf',
```
Comment (RemiBonnet) [inline libs/pages/cluster/src/lib/ui/page-settings-remote/page-settings-remote.spec.tsx:17]: Can you use the mock here ?

**Review by RemiBonnet — APPROVED** [review]
Well done 👍

**Comment by evoxmusic** [comment]
Your preview environment failed to be deployed !
Click on the link below to check the deployment logs.
:point_right: [[PR] staging - feat(cluster): remote step - 2023-02-07T12:18:11Z](https://new.console.qovery.com/organization/141c07c8-0dd9-4623-983b-3fdd61867255/project/977bd069-20f7-4b24-bd95-28b44174cf43/environment/c41fbdb8-201a-461f-b9f8-504bd879fa47/services/logs)

**Pull request merged**